src/Native/System.Native/pal_networkchange.cpp

```diff
@@ -52,7 +52,8 @@ extern "C" NetworkChangeKind SystemNative_ReadSingleEvent(int32_t sock)
     iovec iov = {buffer, sizeof(buffer)};
     sockaddr_nl sanl;
     msghdr msg = {reinterpret_cast<void*>(&sanl), sizeof(sockaddr_nl), &iov, 1, NULL, 0, 0};
-    ssize_t len = recvmsg(sock, &msg, 0);
+    ssize_t len;
+    while (CheckInterrupted(len = recvmsg(sock, &msg, 0)));
```

Comment:
This is consistent with how we do it in the rest of the shims.  Why do you dislike it so much?  I personally prefer:
```C++
while (CheckInterrupted(len = recvmsg(sock, &msg, 0)));
```
over
```C++
do
{
    len = recvmsg(sock, &msg, 0);
}
while (CheckInterrupted(len));
```